internal/handlers/device.go

```diff
@@ -171,20 +171,27 @@ func (api *API) UpdateDevice(c *gin.Context) {
 		}
 
 		if request.ChildPrefix != nil {
-			prefixAllocated := make(map[string]struct{})
-			for _, prefix := range device.ChildPrefix {
-				prefixAllocated[prefix] = struct{}{}
-			}
-			for _, prefix := range request.ChildPrefix {
-				// lookup miss of prefix means we need to release it
-				if _, ok := prefixAllocated[prefix]; ok {
-					if err := api.ipam.ReleasePrefix(ctx, device.OrganizationID, prefix); err != nil {
-						return err
-					}
-				} else {
-					// otherwise we need to allocate it
-					if err := api.ipam.AssignPrefix(ctx, device.OrganizationID, prefix); err != nil {
-						return err
+			// check if the updated device child prefix matches the existing device prefix
+			if !childPrefixEquals(device.ChildPrefix, request.ChildPrefix) {
```

Comment:
Please combine this with the `if` statement that includes this block.

```suggestion
			if request.ChildPrefix != nil && !childPrefixEquals(device.ChildPrefix, request.ChildPrefix) {
```